sansio_lsp_client/client.py

```diff
@@ -171,14 +329,49 @@ def parse_request(event_cls: t.Type[Event]) -> Event:
                     " or ServerNotification"
                 )
 
-        if request.method == "window/showMessage":
+        if request.method == "workspace/workspaceFolders":
+            return parse_request(WorkspaceFolders)
+
+        elif request.method == "workspace/configuration":
+            return parse_request(ConfigurationRequest)
+
+        elif request.method == "window/showMessage":
             return parse_request(ShowMessage)
         elif request.method == "window/showMessageRequest":
             return parse_request(ShowMessageRequest)
         elif request.method == "window/logMessage":
             return parse_request(LogMessage)
+
         elif request.method == "textDocument/publishDiagnostics":
             return parse_request(PublishDiagnostics)
+
+        elif request.method == "window/workDoneProgress/create":
+            assert request.params is not None
+            event = parse_request(WorkDoneProgressCreate)
+            self._progress_tokens_map[request.params["token"]] = WorkDoneProgress
```

Comment:
Why is this a `dict`? It seems to only have `WorkDoneProgress` as a value

Reply:
I cleaned it up. I think halfbrained's original idea was to implement more stuff later, but we can easily add this back if it's needed.